forem/forem
# Runtime Banner v2

## What type of PR is this? (check all applicable)


## Description

Second iteration of the Runtime Banner. Now relying on Preact to implement most of the functionality. The only difference in terms of functionality is that dismissing the banner will keep a record on `localStorage` in order to avoid showing the banner to users that have dismissed it.

## Related Tickets & Documents

https://github.com/forem/rfcs/pull/249

## QA Instructions, Screenshots, Recordings

1. Boot app locally
2. Browse through different pages (main feed, Posts, User pages, etc) on a desktop browser: The banner should not appear visible
3. Browse the app using an iOS device: The banner should appear visible
   - If you don't have an iOS device you can edit [this line](https://github.com/forem/forem/pull/14119/files#diff-454162adb324ee7c994709db71e91a53146f61eaa98f40d524f7d0762e9f1e5aR73) to match a device you do actually have (`Browser-Android`, `Browser-macOS`, `Browser-Windows` or `Browser-Linux`) and the banner should now appear visible
4. Dismiss the banner. Then continue to browse through the app and you should notice the banner is no longer reappearing

### UI accessibility concerns?

None: The banner was refactored but the HTML output should remain the same.

## Added/updated tests?


## [Forem core team only] How will this change be communicated?

- [x] This change does not need to be communicated, and this is why not: It's a bugfix/refactor to an existing feature

## [optional] What gif best describes this PR or how it makes you feel?

![indecisive doge furry](https://media.giphy.com/media/26BoDGStUAafwdZII/giphy.gif)
